### ISSUE TITLE
[ISSUE] 429 Conflicts when deploying multiple notebooks

### ISSUE THREAD

**Issue by damoodamoo** [databrickslabs/terraform-provider-databricks#41]
### Terraform Version
```
Terraform v0.12.24
+ provider.azurerm v1.44.0
+ provider.databricks v0.1.0
```

### Affected Resource(s)
Please list the resources as a list, for example:
- databricks_notebook

### Terraform Configuration Files
```hcl
resource "databricks_notebook" "notebook" {
  for_each = fileset("${path.module}/notebooks", "*")

  content = filebase64("${path.module}/notebooks/${each.value}")
  path = "/Shared/Notebooks/${each.value}"
  overwrite = false
  mkdirs = true
  format = "SOURCE"
  language = SCALA
}
```
The above loops a number of files in a local dir and deploys them to databricks using the `databricks_notebook` provider. When more than about 3/4 files are present i'm seeing pretty regular `429` errors returned when running `tf apply`. Not sure if there's a race condition whereby the notebook is still being deleted when `tf` is trying to re-create it. 

### Debug Output
On first create, all is successful. On further runs of `tf apply` it sees each notebook as needing to be recreated (will log separate issue for this), and when trying to recreate we usually hit a `429` error.
```
databricks_notebook.notebook["notebook1.scala"]: Destroying... [id=/Shared/Notebooks/notebook1.scala]
databricks_notebook.notebook["notebook3.scala"]: Destroying... [id=/Shared/Notebooks/notebook3.scala]
databricks_notebook.notebook["notebook2.scala"]: Destroying... [id=/Shared/Notebooks/notebook2.scala]
databricks_notebook.notebook["notebook3.scala"]: Destruction complete after 0s
databricks_notebook.notebook["notebook3.scala"]: Creating...
databricks_notebook.notebook["notebook2.scala"]: Destruction complete after 0s
databricks_notebook.notebook["notebook2.scala"]: Creating...
databricks_notebook.notebook["notebook3.scala"]: Creation complete after 1s [id=/Shared/Notebooks/notebook3.scala]
databricks_notebook.notebook["notebook2.scala"]: Creation complete after 1s [id=/Shared/Notebooks/notebook2.scala]

Error: Response from server (429) 
```
### Expected Behavior
That the notebooks get re-created.

### Actual Behavior
429 errors

### Steps to Reproduce
1. Create a folder next to your tf script called "notebooks", and add a number of dummy scala files. 
2. Use the above HCL then run `tf apply` to loop the directory and create all the notebooks
3. Run `tf apply` again.

**Comment by stikkireddy** [comment]
@damoodamoo Thanks for pointing this out, hmm when I was testing I was using DBCs but your situation seems like a reasonable ask. So this is a behavior of the workspace notebook delete api being hit concurrently. 

To make notebooks a bit more stable resource, it may be worth doing the following:

1. Separate folders from the notebook resource so that databricks_notebooks & databricks_folders are separate resources. When notebooks are made in a loop in parallel, they try to make folders so duplicate folders in the same level appear and UI looks very glitchy. Separating folders from notebooks will make that better.
2. Put a delay with a random sleep in the delete request if it submits a 429 (https://http.cat/429). To avoid the terraform parallelism to submit delete requests all at once. Behavior seems to only occur on delete requests. Will ask internally about this weird behavior but this seems to be a reasonable way of handling the delete requests. Just have to make sure that there is a random value in the sleep.
3. Add ACC tests for notebooks & folders (folders will only have create, read, delete)
4. Add docs for folders resource.

**Comment by nfx** [comment]
#111
